src/ol_infrastructure/infrastructure/consul/__main__.py

```diff
@@ -219,18 +220,80 @@
         "private_ip": consul_instance.private_ip,
         "ipv6_address": consul_instance.ipv6_addresses,
     }
+######
 
 fifteen_minutes = 60 * 15
 consul_domain = route53.Record(
     f"consul-{environment_name}-dns-record",
-    name=f"consul-{salt_environment}.odl.mit.edu",
+    # name=f"consul-{salt_environment}.odl.mit.edu",
     type="A",
     ttl=fifteen_minutes,
     records=[consul_server.public_ip for consul_server in consul_instances],
     zone_id=mitodl_zone_id,
     opts=ResourceOptions(depends_on=consul_instances),
 )
 
+aws_account = get_caller_identity()
+consul_ami = ec2.get_ami(
+    filters=[
+        ec2.GetAmiFilterArgs(name="name", values=["consul"]),
+        ec2.GetAmiFilterArgs(name="virtualization-type", values=["hvm"]),
+        ec2.GetAmiFilterArgs(name="root-device-type", values=["ebs"]),
+    ],
+    most_recent=True,
+    owners=[aws_account.account_id],
+)
+
+consul_launch_config = ec2.LaunchTemplate(
+    "consul-launch-template",
+    name_prefix=f"consul-{environment_name}-",
+    description="Launch template for deploying Consul cluster",
+    iam_instance_profile=consul_instance_profile.id,
+    image_id=consul_ami.id,
+    instance_type=InstanceTypes[instance_type_name].value,
+    # key_name=salt_config.require("key_name"),
+    tags=instance_tags,
+    tag_specifications=[
+        ec2.LaunchTemplateTagSpecificationArgs(
+            resource_type="instance",
+            tags=aws_config.merged_tags({"Name": f"consul-{stack_info.env_suffix}"}),
+        ),
+        ec2.LaunchTemplateTagSpecificationArgs(
+            resource_type="volume",
+            tags=aws_config.merged_tags({"Name": f"consul-{stack_info.env_suffix}"}),
+        ),
+    ],
+    vpc_security_group_ids=[
+        # destination_vpc["security_groups"]["salt_minion"],
+        destination_vpc["security_groups"]["web"],
+        consul_server_security_group.id,
+    ],
+)
+
+consul_asg = autoscaling.Group(
+    f"consul-{environment_name}-autoscaling-group",
+    availability_zones=availability_zones,
+    desired_capacity=3,
+    max_size=5,
+    min_size=3,
```

Comment:
We'll probably want to set the capacity/max_size/min_size to the same value and pull it from the stack config, defaulting to 3 if it's not present.